### PR TITLE
Optimized plot grouping and added plot buttons to sample table

### PR DESCRIPTION
**Pull request by creilly8**
# Description

These are minor changes we discussed in team meeting some time ago. 
 
Changes:
- When a plot is visible, a button appears alongside the sample in the table, serving as a condensed, visual representation of plots shown. On button click, the page scrolls to that plot. 
- Instead of destroying and recreating plots when groupBy changes, the elements are detached from the parent and reattached to a new parent section. This avoids unnecessary app.dispatches, solves the issue of the half second flickering that would occur, and increases the page response time. 
- Updated `client/plots/sc/README.md` to clarify the SC app's grouping logic, especially for plot data fetching (`checkPlotAvailability`) and section grouping by `groupBy` setting. The documentation now accurately describes how plot buttons and subplots are managed and how sections are regrouped without destroying plot components. 
-  Updated `SCViewModel.unit.spec.ts` tests to reflect new columns in the sample table (such as the "Shown plots" column), adjusting expectations for the number and order of columns and the mapping of row values. 
- Improved `SectionRenderer.unit.spec.ts` test descriptions and coverage to include new grouping logic and section removal behavior, ensuring tests match the updated implementation. 
- Added a full unit test suite for `SampleTableRenderer` in `SampleTableRenderer.unit.spec.ts`, covering constructor behavior, table rendering, row selection, plot button rendering, and edge cases such as missing IDs and long plot names. This ensures robust coverage of table interactions and UI updates.

Test with any of the SC app examples from http://localhost:3000/url.html. 

This work supports efforts described in the [SC roadmap](https://github.com/stjude/sjpp/issues/658)

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.

- [x] Tests: Added and/or passed unit and integration tests, or N/A
- [x] Todos: Commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A
- [x] Rust: Checked to see whether Rust needs to be re-compiled because of this PR, or N/A

### CODE REVIEW

File: client/plots/sc/viewModel/SCViewModel.ts [hunk 1]
```diff
@@ -43,7 +43,10 @@ export class SCViewModel {
 		const hasExperiments = items.some(i => i.experiments)
 
 		// first column is sample and is hardcoded
-		const columns: TableColumn[] = [{ label: plotConfig.settings.sc.columns.sample, sortable: true }]
+		const columns: TableColumn[] = [
+			{ label: plotConfig.settings.sc.columns.sample, sortable: true },
+			{ label: 'Shown plots' }
+		]
 		if (hasExperiments) columns.push({ label: 'Sample', sortable: true }) //add after the case column
```
[the remark this answers is not in the document]
Comment (creilly8) [reply client/plots/sc/viewModel/SCViewModel.ts:54]: Fixed in https://github.com/stjude/proteinpaint/pull/4540/commits/6c937bbca6f74c708baa8cc52ca376317176bd49

File: client/plots/sc/view/SampleTableRenderer.ts [hunk 2]
```diff
@@ -43,10 +47,49 @@ export class SampleTableRenderer {
 					key = key === 'sample' ? 'sID' : key === 'experiment' ? 'eID' : key
 					item[key] = r.value
 				})
+				this.rowIndex = index
 				if (!item.sID) throw new Error('Selected item must have sID property')
 				this.interactions.updateItem(item)
 				this.dom.plotsBtnsDiv.style('display', 'block')
```
[the remark this answers is not in the document]
Comment (creilly8) [reply client/plots/sc/view/SampleTableRenderer.ts:44]: Resolved in https://github.com/stjude/proteinpaint/pull/4540/commits/6c937bbca6f74c708baa8cc52ca376317176bd49

**Review by xzhou82 — APPROVED** [review]

**Comment by xzhou82** [comment]
a deleted plot icon can still remain in Shown Plots
[link](http://localhost:3000/?mass={%22genome%22:%22hg38%22,%22dslabel%22:%22BALL-scrna%22,%22nav%22:{%22header_mode%22:%22hidden%22},%22plots%22:[{%22chartType%22:%22sc%22}]}), select sample1 and show umap, select sample2 and show umap. then delete both sandboxes. umap icon of 1st sample remains in table
